### PR TITLE
migrating golangci-lint to v1_30_0 and enabling some new linters

### PR DESCRIPTION
**Pull request by jgheewala**

### CODE REVIEW

File: metadata/hostmetadata/host.go [hunk 1]
```diff
@@ -122,9 +124,11 @@ func int8ArrayToByteArray(in []int8) []byte {
 }
 
 // GetOS returns a struct with information about the host os
-func GetOS() (info *OS, err error) {
-	info = &OS{}
-	hInfo, err := hostInfo()
+func GetOS() (*OS, error) {
+	var (
```
Comment (mdubbyap) [inline metadata/hostmetadata/host.go:128]: this was required?

File: sfxclient/wrapper.go [hunk 2]
```diff
@@ -17,6 +17,7 @@ func NewMultiCollector(collectors ...Collector) Collector {
 }
 
 // Datapoints returns the datapoints from every collector.
+// nolint:prealloc
```
Comment (mdubbyap) [inline sfxclient/wrapper.go:20]: why not pre alloc size of m? likely they'll be at least 1 dp per?

File: web/bucketreqcounter.go [hunk 3]
```diff
@@ -15,8 +15,10 @@ type BucketRequestCounter struct {
 	Bucket            *sfxclient.RollingBucket
 }
 
-var _ HTTPConstructor = (&BucketRequestCounter{}).Wrap
-var _ NextHTTP = (&BucketRequestCounter{}).ServeHTTP
+var (
```
Comment (mdubbyap) [inline web/bucketreqcounter.go:18]: i'm not sure i like this more than 1 requires a var() block...sigh

File: metadata/hostmetadata/host.go (hunk 1 above)
Comment (jgheewala) [reply metadata/hostmetadata/host.go:128]: it was complaining about shadow error so i had to create a var hInfo and if I do that it gofumtd would have complained.. So just merged all var into one declaration

File: sfxclient/wrapper.go (hunk 2 above)
Comment (jgheewala) [reply sfxclient/wrapper.go]: done

**Pull request merged**